README.md
# Simple Form Builder
[![pub package](https://img.shields.io/pub/v/simple_form_builder.svg)](https://pub.dev/packages/simple_form_builder)
[![GitHub Stars](https://img.shields.io/github/stars/tanmoy27112000/SimpleFormBuilder.svg?logo=github)](https://pub.dev/packages/simple_form_builder)
[![Platform](https://img.shields.io/badge/Platform-Android%20%7C%20IOS%20%7C%20Web-green)](https://img.shields.io/badge/Platform-Android%20%7C%20IOS%20%7C%20Web-green)

A simple, colorful and fluid bottom navigation built with flutter

Maintainer : [Tanmoy Karmakar](https://tanmoykarmakar.in)<br>


### Specs
<!-- [![pub](https://img.shields.io/pub/v/flash.svg?style=flat)](https://pub.dev/packages/flash) -->


This library allows you easily implement bottom navigation with many customizable fields to make it your own
<!-- multiple types of fields `text` , `checkbox`, `multiselect` , `datetime` , `date` , `time` , and `file upload`.
This package also provides additional remark options. -->

It has been written **100% in Dart**. ❤️

<p>
  <img width="205px" alt="Example" src="https://i.imgur.com/3uxKgfr.mp4"/>
</p>


<br>

## Installing
Add the following to your `pubspec.yaml` file:
```yaml
dependencies:
  simple_bottom_nav: ^0.0.2
```

<br>

## Sample code

```
import 'package:flutter/material.dart';
import 'package:simple_bottom_nav/simple_bottom_nav.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Material App',
      home: Scaffold(
        body: SimpleBottomNavigator(
          navWidgetList: [
            BottomNavModel(
              title: "Home",
              index: 0,
              activeColor: const Color(0xffFB65B2),
              iconList: [
                "assets/Component 3.png",
                "assets/Home.png",
              ],
              context: context,
            ),
            BottomNavModel(
              title: "Recipe",
              index: 1,
              activeColor: const Color(0xff2EB2FF),
              iconList: [
                "assets/foodtrayFilled.png",
                "assets/foodtray.png",
              ],
              context: context,
            ),
          ],
          pageList: const [
            Scaffold(
              body: Center(
                child: Text("Home"),
              ),
            ),
            Scaffold(
              body: Center(
                child: Text("Recipe"),
              ),
            ),
          ],
        ),
      ),
    );
  }
}

```

<br>

## Simple Usage

To integrate the simple bottom navigation bar just implement it with the following code


## Custom Usage
There are several options that allow for more control:

|  Properties  |   Description   |
|--------------|-----------------|
| `navWidgetList` | This takes a list of BottomNavModel object and converts them to bottom navigation widget |
| `pageList` | This takes in a list of widget preferably Scaffold to show corresponding screens |
<br>

CHANGELOG.md
## 0.0.1

* Initial release

## 0.0.2

* readme added, example improved


lib/src/bottom_nav.dart
import 'dart:io';

import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import 'package:simple_bottom_nav/global/bottom_nav_model.dart';
import 'package:simple_bottom_nav/simple_bottom_nav.dart';
import 'package:simple_bottom_nav/src/page_controller.dart';

class SimpleBottomNavigator extends StatefulWidget {
  final List<Widget> pageList;
  final List<BottomNavModel>? navWidgetList;

  const SimpleBottomNavigator({
    Key? key,
    required this.pageList,
    required this.navWidgetList,
  }) : super(key: key);

  @override
  _SimpleBottomNavigatorState createState() => _SimpleBottomNavigatorState();
}

class _SimpleBottomNavigatorState extends State<SimpleBottomNavigator> {
  @override
  void initState() {
    super.initState();
  }

  @override
  void dispose() {
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return ChangeNotifierProvider(
      create: (context) => PageViewController(),
      builder: (context, child) => Consumer<PageViewController>(
        builder: (context, controller, child) {
          return Scaffold(
            backgroundColor: Colors.transparent,
            body: PageView(
              onPageChanged: (value) {
                controller.setPageIndex(value);
              },
              controller: controller.pageController,
              children: widget.pageList,
            ),
            bottomNavigationBar: Container(
              height: Platform.isIOS ? 100 : 80,
              decoration: const BoxDecoration(
                borderRadius: BorderRadius.only(
                  topLeft: Radius.circular(30),
                  topRight: Radius.circular(30),
                ),
                color: Colors.white,
              ),
              child: Row(
                mainAxisAlignment: MainAxisAlignment.spaceEvenly,
                children: widget.navWidgetList!
                    .map((e) => bottomNavWidget(
                          context: e.context,
                          activeColor: e.activeColor,
                          controller: controller,
                          iconList: e.iconList,
                          index: e.index,
                          isSelected: e.isSelected,
                          title: e.title,
                        ))
                    .toList(),
                // [
                //   bottomNavWidget(
                //     title: "Home",
                //     index: 0,
                //     activeColor: const Color(0xffFB65B2),
                //     controller: controller,
                //     iconList: [
                //       "Component 3",
                //       "Home",
                //     ],
                //     context: context,
                //   ),
                //   bottomNavWidget(
                //     title: "Recipe",
                //     index: 1,
                //     activeColor: const Color(0xff2EB2FF),
                //     controller: controller,
                //     iconList: [
                //       "foodtrayFilled",
                //       "foodtray",
                //     ],
                //     context: context,
                //   ),
                //   bottomNavWidget(
                //     title: "Pregency",
                //     index: 2,
                //     activeColor: const Color(0xff35B729),
                //     controller: controller,
                //     iconList: [
                //       "Group 526",
                //       "Frame 9",
                //     ],
                //     context: context,
                //   ),
                //   bottomNavWidget(
                //     title: "Notification",
                //     index: 3,
                //     activeColor: const Color(0xffB760E6),
                //     controller: controller,
                //     iconList: [
                //       "Component 3-2",
                //       "Component 4",
                //     ],
                //     context: context,
                //   ),
                //   bottomNavWidget(
                //     title: "Profile",
                //     index: 4,
                //     activeColor: const Color(0xff707070),
                //     controller: controller,
                //     iconList: [
                //       "Component 3-3",
                //       "Component 6",
                //     ],
                //     context: context,
                //   ),
                // ],
              ),
            ),
          );
        },
      ),
    );
  }
}


lib/global/bottom_nav_widget.dart
import 'package:flutter/material.dart';
import 'package:simple_bottom_nav/src/page_controller.dart';

Widget bottomNavWidget({
  bool isSelected = false,
  int index = 0,
  Color? activeColor,
  List? iconList,
  PageViewController? controller,
  String title = "Recipe",
  required BuildContext context,
}) {
  return InkWell(
    onTap: () {
      controller!.setPageIndex(index);
      controller.pageController.jumpToPage(
        index,
      );
    },
    child: Column(
      mainAxisAlignment: MainAxisAlignment.center,
      children: [
        Container(
          decoration: BoxDecoration(
            color: controller!.currentPageIndex == index
                ? activeColor
                : Colors.transparent,
            shape: BoxShape.circle,
          ),
          height: 40,
          width: 40,
          child: Padding(
            padding: const EdgeInsets.all(8.0),
            child: Image(
              image: controller.currentPageIndex == index
                  ? AssetImage(iconList![0])
                  : AssetImage(iconList![1]),
            ),
          ),
        ),
        Padding(
          padding: const EdgeInsets.all(2.0),
          child: Text(
            title,
            style: TextStyle(
              fontSize: 13,
              fontWeight: controller.currentPageIndex == index
                  ? FontWeight.bold
                  : FontWeight.normal,
            ),
          ),
        ),
      ],
    ),
  );
}


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:simple_bottom_nav/simple_bottom_nav.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Material App',
      home: Scaffold(
        body: SimpleBottomNavigator(
          navWidgetList: [
            BottomNavModel(
              title: "Home",
              index: 0,
              activeColor: const Color(0xffFB65B2),
              iconList: [
                "assets/Component 3.png",
                "assets/Home.png",
              ],
              context: context,
            ),
            BottomNavModel(
              title: "Recipe",
              index: 1,
              activeColor: const Color(0xff2EB2FF),
              iconList: [
                "assets/foodtrayFilled.png",
                "assets/foodtray.png",
              ],
              context: context,
            ),
            BottomNavModel(
              title: "Pregency",
              index: 2,
              activeColor: const Color(0xff35B729),
              iconList: [
                "assets/Group 526.png",
                "assets/Frame 9.png",
              ],
              context: context,
            ),
            BottomNavModel(
              title: "Profile",
              index: 3,
              activeColor: const Color(0xff707070),
              iconList: [
                "assets/Component 3-3.png",
                "assets/Component 6.png",
              ],
              context: context,
            ),
          ],
          pageList: const [
            Scaffold(
              backgroundColor: Colors.pink,
              body: Center(
                child: Text("Home"),
              ),
            ),
            Scaffold(
              backgroundColor: Colors.blue,
              body: Center(
                child: Text("Recipe"),
              ),
            ),
            Scaffold(
              backgroundColor: Colors.greenAccent,
              body: Center(
                child: Text("Pregnency"),
              ),
            ),
            Scaffold(
              backgroundColor: Colors.grey,
              body: Center(
                child: Text("Profile"),
              ),
            ),
          ],
        ),
      ),
    );
  }
}
